### DISCUSSION TITLE
Connection problem to backend

### DISCUSSION THREAD

**Discussion by aksebzeci** [ikhsan3adi/AuctionApp#8] [Q&A] [answered]
Hello,

I run backend on localhost. When I run app on I get authentication errors. Also, I get internet connection error in registration screen.

Here is my .env file:
API_BASE_URL = 'http://localhost:8000/app/'

Also, I tried these links in API_BASE_URL too. But It didn't work.
'http://localhost:8000/api/'
'http://localhost:8000/'

When the app load I get errors on console log:
 flutter: AuthBloc AuthAppStartedEvent()
flutter: AuthBloc Transition { currentState: AuthStateInitial(), event: AuthAppStartedEvent(), nextState: AuthStateLoading() }
flutter: AuthBloc Change { currentState: AuthStateInitial(), nextState: AuthStateLoading() }
flutter: AuthBloc type 'String' is not a subtype of type 'int' of 'index'
flutter: AuthBloc _AuthStatusChangedEvent(Instance of 'AuthUnknown')
flutter: AuthBloc Transition { currentState: AuthStateLoading(), event: _AuthStatusChangedEvent(Instance of 'AuthUnknown'), nextState: AuthStateUnknown([]) }
flutter: AuthBloc Change { currentState: AuthStateLoading(), nextState: AuthStateUnknown([]) }

The error in registration screen:
![image](https://github.com/ikhsan3adi/flutter-auction-app/assets/79276966/3fd07f6c-7d88-4df7-8e56-e8ee78c183f0)

**Comment by aksebzeci** [comment]
RegisterBloc Change { currentState: RegisterState(blob:http://localhost:58426/de17258b-a909-4541-b972-dca2120bf982, Instance of 'RegisterFormState', true, Expected a value of type 'String', but got one of type 'Null'), nextState: RegisterState(blob:http://localhost:58426/de17258b-a909-4541-b972-dca2120bf982, Instance of 'RegisterFormState', true, Expected a value of type 'String', but got one of type 'Null') }
RegisterBloc Transition { currentState: RegisterState(blob:http://localhost:58426/de17258b-a909-4541-b972-dca2120bf982, Instance of 'RegisterFormState', true, Expected a value of type 'String', but got one of type 'Null'), event: AttemptRegister(), nextState: RegisterState(blob:http://localhost:58426/de17258b-a909-4541-b972-dca2120bf982, Instance of 'RegisterFormState', true, Unsupported operation: MultipartFile is only supported where dart:io is available.) }
RegisterBloc Change { currentState: RegisterState(blob:http://localhost:58426/de17258b-a909-4541-b972-dca2120bf982, Instance of 'RegisterFormState', true, Expected a value of type 'String', but got one of type 'Null'), nextState: RegisterState(blob:http://localhost:58426/de17258b-a909-4541-b972-dca2120bf982, Instance of 'RegisterFormState', true, Unsupported operation: MultipartFile is only supported where dart:io is available.) }

 
![image](https://github.com/ikhsan3adi/flutter-auction-app/assets/79276966/43cbf89a-4213-497a-b6aa-9bda1718d96f)

**Reply by ikhsan3adi** [reply]
Are you running the app in a browser?

**Comment by ikhsan3adi** [comment, marked as answer]
I have fixed the type casting error on the register page

```dart
@override
  Future<void> register(User user, String password) async {
    FormData formData = FormData();

    if (user.profileImageUrl != null) {
      formData.files.add(MapEntry(
        "profile_image",
        await MultipartFile.fromFile(user.profileImageUrl!, filename: user.profileImageUrl!.split('/').last),
      ));
    }

    // i added this line
    Map<String, dynamic> newUser = user.toJson().map((key, value) => MapEntry(key, value ?? 'null'));

    formData.fields.addAll(newUser.cast<String, String>().entries);
    formData.fields.add(MapEntry('password', password));

    await _dio.post(
      '/users',
      data: formData,
    );
  }
```

Check the latest commit

**Comment by ikhsan3adi** [comment]
I've tried using `localhost:8080` (using php spark serve) but it doesn't work.
My suggestion is to use local hosting using a webserver like apache or XAMPP.

**Comment by aksebzeci** [comment]
Hello,I run backend on Chrome. I run the auction app on Ios and Chrome.On local php and composer installed.I will try apache on local machine.Mustafa AksebzeciOn 11 Aug 2023, at 15:38, Ikhsan Satriadi ***@***.***> wrote:﻿
I've tried using localhost:8080 (using php spark serve) but it doesn't work.
My suggestion is to use local hosting using a webserver like apache or XAMPP.

**Comment by aksebzeci** [comment]
Hello again,

I setup local and remote backend servers. But I couldn't get it run. Can you use this url and debug the problem?

API_BASE_URL = 'https://mezatla-backend-aaebd516df80.herokuapp.com/'

Also, I get this error in login screen:

 LoginBloc type 'String' is not a subtype of type 'int' of 'index'
